plugins/module_utils/omevv_utils/omevv_firmware_utils.py

```diff
@@ -254,3 +274,337 @@ def delete_firmware_repository_profile(self, profile_id):
+        """
+        Retrieves the cluster information.
+
+        Args:
+            vcenter_uuid: UUID of the vCenter.
+
+        Returns:
+            list: The list of all cluster information.
+        """
+
+        resp = self.omevv.invoke_request('GET', CLUSTER_URI.format(vcenter_uuid=vcenter_uuid))
+        return resp
+
+    def get_cluster_id(self, cluster_names, vcenter_uuid):
+        """
+        Fetch cluster IDs for the given clusters.
+
+        Args:
+            vcenter_uuid: UUID of the vCenter.
+            cluster_names: List of cluster names to fetch group IDs for.
+
+        Returns:
+            list: A list of cluster IDs.
+        """
+        clusters_resp = self.omevv.invoke_request('GET', CLUSTER_URI.format(vcenter_uuid=vcenter_uuid))
+        clusters = clusters_resp.json_data if clusters_resp.success else []
+
+        # Map cluster name to entity ID (clustId)
+        cluster_ids = [c['entityId'] for c in clusters if c['name'] in cluster_names]
+
+        return cluster_ids
+
+    def get_group_ids_for_clusters(self, vcenter_uuid, cluster_names):
+
+        """
+        Fetch group IDs for the given clusters.
+
+        Args:
+            vcenter_uuid: UUID of the vCenter.
+            cluster_names: List of cluster names to fetch group IDs for.
+
+        Returns:
+            list: A list of group IDs.
+        """
+        clusters_resp = self.omevv.invoke_request('GET', CLUSTER_URI.format(vcenter_uuid=vcenter_uuid))
+        clusters = clusters_resp.json_data if clusters_resp.success else []
+
+        # Map cluster name to entity ID (clustId)
+        cluster_ids = [c['entityId'] for c in clusters if c['name'] in cluster_names]
+
+        # Fetch group IDs for the identified cluster IDs
+        group_ids = []
+        if cluster_ids:
+            payload = {"clustIds": cluster_ids}
+            group_resp = self.omevv.invoke_request('POST', CLUSTER_IDS_URI.format(vcenter_uuid=vcenter_uuid), payload)
+            group_ids = [g['groupId'] for g in group_resp.json_data] if group_resp.success else []
+
+        return group_ids
+
+    def get_repo_id_by_name(self, repository_profile):
+        repo_profile_info = self.omevv_profile_obj.get_firmware_repository_profile(
+            profile_name=repository_profile
+        )
+
+        firmware_repo_id = None
+        if repo_profile_info:
+            firmware_repo_id = repo_profile_info.get("id")
+        return firmware_repo_id
+
+    def get_baseline_profiles(self, vcenter_uuid):
+        """
+        Retrieves the list of baseline profiles associated with a given vCenter.
+
+        Args:
+            vcenter_uuid (str): The UUID of the vCenter server.
+
+        Returns:
+            list: A list of baseline profiles.
+
+        Raises:
+            None
+        """
+        response = self.omevv.invoke_request('GET', BASELINE_PROFILE_URI.format(vcenter_uuid=vcenter_uuid))
+        if response.success:
+            return response.json_data
+        else:
```

Comment:
We can remove else and directly return []

Reply:
Done